### PR TITLE
Add ability to change ordering of sections

### PR DESCRIPTION
**Pull request by anishg24**
**TL;DR** You can define a custom ordering of sections within the `meta.order` field of your `resume.json` like below:
```json
{
  "meta": {
      "theme": "even",
      "colors": {
        "background": ["#ffffff", "#191e23"],
        "dimmed": ["#f3f4f5", "#23282d"],
        "primary": ["#191e23", "#fbfbfc"],
        "secondary": ["#6c7781", "#ccd0d4"],
        "accent": ["#0073aa", "#00a0d2"]
      },
      "order": [
        "work",
        "volunteer",
        "publications",
        "education",
        "projects",
        "awards",
        "certificates",
        "skills",
        "languages",
        "interests",
        "references"
      ]
    }
}
```

My reasoning for this change that I didn't like the default ordering of the theme, I wanted to make some sections appear first before others. Thanks to the modularity of this theme, it was a relatively simple fix. This fix shouldn't affect any current users, as if no order supplied, the default one provided by the maintainers will be used.

This fix also means that you can choose to omit certain sections altogether or (if you want) display the same section again and again.

Because of a little weirdness in newlines generating, I had to provide a new snapshot file, but you can see that the changes are minimal and only apply to a couple of newlines.

I have tested these changes to a success on my local resume, the sample resume, as well as if the order has an invalid/misspelt selection. By default, if the order contains a section that doesn't exist, it simply ignores it. 

Lastly, I have modularized the code so that it fits the overall code-style of this repository.

Please let me know if you have any other questions, and hats off to the team for making these edits quite simple.

### CODE REVIEW

File: components/body.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        certificates: Certificates(resume.certificates),
+        publications: Publications(resume.publications),
+        skills: Skills(resume.skills),
+        languages: Languages(resume.languages),
+        interests: Interests(resume.interests),
+        references: References(resume.references),
+    }
+
+    return html`${resumeOrder.map(section => html`${orderComponentMap[section]}`)}`;
+}
```
Comment (rbardini) [inline components/body.js]: We can simplify things a bit here by using the component map order as the default section order, while also deferring section rendering until it's needed (plus some renaming/null-checks):

```suggestion
const sectionComponentMap = {
  work: (resume) => Work(resume.work),
  volunteer: (resume) => Volunteer(resume.volunteer),
  education: (resume) => Education(resume.education),
  projects: (resume) => Projects(resume.projects),
  awards: (resume) => Awards(resume.awards),
  certificates: (resume) => Certificates(resume.certificates),
  publications: (resume) => Publications(resume.publications),
  skills: (resume) => Skills(resume.skills),
  languages: (resume) => Languages(resume.languages),
  interests: (resume) => Interests(resume.interests),
  references: (resume) => References(resume.references),
}

export default function Sections(resume) {
  const sections = resume.meta?.sections || Object.keys(sectionComponentMap)
  return html`${sections.map(section => sectionComponentMap[section]?.(resume))}`
}

```

File: resume.js [hunk 2]
```diff
@@ -1,18 +1,8 @@
...
 import Meta from './components/meta.js'
-import Projects from './components/projects.js'
-import Publications from './components/publications.js'
-import References from './components/references.js'
-import Skills from './components/skills.js'
-import Volunteer from './components/volunteer.js'
-import Work from './components/work.js'
 import colors from './utils/colors.js'
 import html from './utils/html.js'
+import Body from "./components/body.js";
```
Comment (rbardini) [inline resume.js]: I think we should rename this component `Sections`, since it doesn't render the entire `body` of the page—`Header` is also rendered there (plus reordering imports):

```suggestion
import Sections from './components/sections.js'
import colors from './utils/colors.js'
import html from './utils/html.js'
```

File: README.md [hunk 3]
```diff
@@ -73,3 +73,31 @@ Here's an example using the default theme colors:
...
+### Order
+
+You can override the default order of sections via the `.meta.order` resume field.
+You can put however many sections you wish, ordered from top to bottom.
+The available sections you may order are provided below:
+
+```json
+{
+  "meta": {
+    "order": [
```
Comment (rbardini) [inline README.md]: I think we should rename `.meta.order` to `.meta.sections`, since it not only defines the order, but also what sections are actually displayed (plus some rewording suggestions):

```suggestion
### Sections

You can override what sections are displayed, and in what order, via the `.meta.sections` resume field.

Here's an example with all available sections in their default order:

```json
{
  "meta": {
    "sections": [
```

**Review by rbardini — COMMENTED** [review]
Thanks for your contribution, @anishg24, I really like the idea! Left some suggestions:

**Comment by anishg24** [comment]
Great optimizations @rbardini, I forget the power of JavaScript sometimes. Thank you for taking the time to review!

File: components/body.js
Comment (rbardini) [inline components/body.js:1]: We must rename this file `setions.js` to satisfy the import.

**Review by rbardini — COMMENTED** [review]
Almost there! Build is failing with `Could not resolve ./components/sections.js from resume.js` 👇

**Comment by cometsong** [comment]
@anishg24 - Is this PR still active for you?
If not, I'll take it on and make the needed changes.
This mod of section ordering is an *excellent* idea!

**Comment by anishg24** [comment]
@cometsong Unfortunately my courses started to kick it up a notch so I don't think I can finish this until my next break, but you are more than welcome to make the remaining changes!

**Comment by cometsong** [comment]
I've put the fixes to this into a new PR #20 😃

**Comment by rbardini** [comment]
Closing in favor of #20.
